igor-gateway/examples/zwave.rs

```diff
@@ -0,0 +1,54 @@
+use std::io;
+use std::io::Write;
+use igor_gateway::zwave::ZWaveController;
+
+fn display_prompt() {
+    print!("> ");
+    io::stdout().flush().unwrap(); // https://github.com/rust-lang/rust/issues/23818
+}
+
+fn main() {
+
+    let controller = ZWaveController::new(String::from("/dev/cu.usbmodem141401"));
+
+    println!("Enter `exit` or Control-C to exit.");
+    let mut input = String::new();
+    loop {
+        input.clear();
+        display_prompt();
+
+        if let Ok(n) = io::stdin().read_line(&mut input) {
+            if n == 0 {
+                // End-of-file (either Control-D or we were redirected).
+                break;
+            }
+        } else {
+            println!("Error reading stdin");
+            break;
+        }
+
+        let tokens: Vec<_> = input.split_whitespace().collect();
+        if tokens.is_empty() {
+            continue;
+        }
+
+        println!("read token: {}", tokens[0]);
+
+        match tokens[0] {
```

Comment:
Since you only care about first token:
```
let mut tokens =  input.split_whitespace();
if let Some(token) =  tokens.next() {
 yo match here
}
```
Saves you a whole allocation.

Reply:
but i think i will eventually care about all tokens e.g., "set <nodeId> <instruction>"

Reply:
We will use https://github.com/TeXitoi/structopt for actual CLI. Just showing an easier way to what you're doing here. No need to change it in example.